### PR TITLE
Changes to the UI: Updated LAO screens including Identity, Notification and Events

### PR DESCRIPTION
**Pull request by Tyratox**
This reworks most of the remaining screens that were not covered by #1038.
There is still a lot of work that needs to be done but this PR tries to make the UI consistent and somewhat clean. Making it fancy would require more work.

## Lao List
Screenshot 2022-05-28 at 10 22 39

## Lao Home
Screenshot 2022-05-28 at 10 22 48

Screenshot 2022-05-28 at 10 22 51

## Notifications
Screenshot 2022-05-28 at 10 23 00

## Events
Screenshot 2022-05-28 at 10 24 05

Screenshot 2022-05-28 at 10 23 21
Screenshot 2022-05-28 at 10 23 35
Screenshot 2022-05-28 at 10 23 57

Screenshot 2022-05-28 at 10 26 43

## Roll Call
Screenshot 2022-05-28 at 10 26 54
Screenshot 2022-05-28 at 10 26 57
Screenshot 2022-05-28 at 10 27 00

 
Screenshot 2022-05-28 at 10 27 32
Screenshot 2022-05-28 at 10 27 37

Screenshot 2022-05-28 at 10 27 43

## Election
Screenshot 2022-05-28 at 10 27 59
Screenshot 2022-05-28 at 10 28 01

Screenshot 2022-05-28 at 10 28 39
Screenshot 2022-05-28 at 10 28 45

### CODE REVIEW

File: fe1-web/.eslintrc.js
Comment (Tyratox) [inline fe1-web/.eslintrc.js:107]: Unfortunately this leads to false positives if some other component already wraps its children in a <Text/>.

File: fe1-web/src/features/rollCall/screens/ViewSingleRollCall.tsx
Comment (Tyratox) [inline fe1-web/src/features/rollCall/screens/ViewSingleRollCall.tsx:2]: I think this file should in a future PR be split as in the case of the election

File: fe1-web/src/features/notification/screens/NotificationScreen.tsx
Comment (Xelowak) [inline fe1-web/src/features/notification/screens/NotificationScreen.tsx:103]: I feel like it is very similar for both notification types. It could be nice to create a function to remove this duplication if it doesn't introduce too many arguments.

File: fe1-web/src/features/rollCall/screens/ViewSingleRollCall.tsx
Comment (Xelowak) [reply fe1-web/src/features/rollCall/screens/ViewSingleRollCall.tsx:2]: Yes that's a great idea given the number of lines.

File: fe1-web/src/features/rollCall/screens/ViewSingleRollCall.tsx
Comment (Xelowak) [inline fe1-web/src/features/rollCall/screens/ViewSingleRollCall.tsx:297]: Since these strings will be displayed in the UI (toast message), I guess you can add them in `string.ts`.

File: fe1-web/src/features/rollCall/screens/__tests__/ViewSingleRollCall.test.tsx
Comment (Xelowak) [inline fe1-web/src/features/rollCall/screens/__tests__/ViewSingleRollCall.test.tsx:76]: For your information, you can do this by setting the local time of the system, like in [here](https://github.com/dedis/popstellar/blob/master/fe1-web/src/features/social/components/__tests__/ChirpCard.test.tsx).

**Review by Xelowak — COMMENTED** [review]
Thanks a lot for all of this! I added a few comments but overall this is great work.

I'd suggest adding descriptive comments at the beginning of all files you added (like `ElectionListItem`), so that it's easier to understand what it does and how to re-use it if needed. It can be relevant in files that contain multiple components, too.

I've not tested the UI myself so I trust you about the fact that everything works fine. However, I saw on your 5th screenshot (about events) that there is a dollar sign before the date and I imagine it's not intended.

Maybe it could be nice to check if you can create private functions or components in some classes/components, because I feel like the code you wrote (especially those `.map`) is very similar in certain places.

Given the amount of code to review, I'm quite sure that I didn't catch everything but since the quality of your code is really good it should be ok :). If something breaks, you still have time to fix it until the final deadline anyway.

File: fe1-web/src/features/rollCall/screens/__tests__/ViewSingleRollCall.test.tsx
Comment (Tyratox) [reply fe1-web/src/features/rollCall/screens/__tests__/ViewSingleRollCall.test.tsx:76]: I don't think this will work. While this will set the time, it will not make the output consistent. `toLocale...` functions depend on the browser / node settings which means the CI can suddenly output something else and the test will break.

File: fe1-web/src/features/evoting/components/ElectionTerminated.tsx
Comment (Tyratox) [reply fe1-web/src/features/evoting/components/ElectionTerminated.tsx:11]: Yes I think a lot of the duplication will go away by creating a ElectionQuestion component 👍

File: fe1-web/src/features/evoting/screens/ViewSingleElection.tsx
Comment (Tyratox) [reply fe1-web/src/features/evoting/screens/ViewSingleElection.tsx:73]: Oh oops, thank you!! 😬

File: fe1-web/src/features/home/interface/Configuration.ts
Comment (Tyratox) [reply fe1-web/src/features/home/interface/Configuration.ts:87]: 👍

File: fe1-web/src/features/home/objects/ConnectToLao.ts
Comment (Tyratox) [reply fe1-web/src/features/home/objects/ConnectToLao.ts:41]: Hmmm..  Apparently QR codes can hold up to 3 KB of information. I guess we would need to compute the size in bytes to throw an accurate warning as it will also depend on the length of the server addresses.. I think this should be doable using Buffers 🤔

File: fe1-web/src/features/notification/screens/NotificationScreen.tsx
Comment (Tyratox) [reply fe1-web/src/features/notification/screens/NotificationScreen.tsx:103]: I'll try it :)

File: fe1-web/src/features/home/objects/ConnectToLao.ts
Comment (Tyratox) [reply fe1-web/src/features/home/objects/ConnectToLao.ts:41]: Hmm okay apparently it is not so easy because it also depends on the error correction level and the version used 😬

**Comment by Tyratox** [comment]
Thank you for the review! 😁

> I'd suggest adding descriptive comments at the beginning of all files you added (like ElectionListItem), so that it's easier to understand what it does and how to re-use it if needed. It can be relevant in files that contain multiple components, too.

On it!

> I've not tested the UI myself so I trust you about the fact that everything works fine. However, I saw on your 5th screenshot (about events) that there is a dollar sign before the date and I imagine it's not intended.

Yes, I realized this too. The screenshots are not up to date since there were a few more issues I noticed. 😅

> Maybe it could be nice to check if you can create private functions or components in some classes/components, because I feel like the code you wrote (especially those .map) is very similar in certain places.

I just spent 1.5 hours trying to generalize the list component but unfortunately it is not trivial at all without sacrificing generality. (I wanted to get rid of the `List.getListItemStyles(..)` but this requires a general component to know the number of times which means we have to pass an array of components to it instead of of simple react children. This then again means we cannot apply the `key` property without a lot of react trickery and magic which probably is not stable 😅)

> Given the amount of code to review, I'm quite sure that I didn't catch everything but since the quality of your code is really good it should be ok :). If something breaks, you still have time to fix it until the final deadline anyway.

Haha yes! :) I will make sure to test all actions in the UI before the deadline!!

**Comment by Tyratox** [comment]
I just added comments to these files :)

**Comment by Tyratox** [comment]
Will fix #982 by replacing the graph with showing the number of votes for each option. In the future it is possible to re-introduce a graph but readability should be considered this time around.

**Review by Xelowak — APPROVED** [review]
I've had a quick look at everything you changed and it seems fine to me.

Regarding the function that I suggested to add, even if it's not possible to re-use them across files, maybe it could be nice to split a little bit the code when the return is large ( like in `ElectionOpened`). It would make it clearer to understand which part renders which part of the component/screen instead of having a lot of code directly in the return. 

I'm already approving if you want to treat it later, but I think it would be great.
You still have failing tests though, but I don't know why.

**Pull request merged**
